### ISSUE TITLE
Stuck at "Torrent waiting for file selection"

### ISSUE THREAD

**Issue by minus1kelvin** [rogerfar/rdt-client#50]
Hello. I seem to be having this issue intermittently  and was wondering how others resolve. I saw this was an issue awhile ago, but it just stopped for the user.

**Comment by rogerfar** [comment]
Are you running the latest version? As far as I know this shouldn't happen anymore.

If it does, can you enable logging to verbose, then send the logfile in a pastebin or attachment?

**Comment by janitory** [comment]
This issue reappeared for me with the latest versions.

Log entry: 
```
2021-06-14 19:13:25.728 +02:00 [DBG] TorrentRunner Tick Start
2021-06-14 19:13:25.728 +02:00 [DBG] Currently 0 active downloads
2021-06-14 19:13:25.729 +02:00 [DBG] Currently 0 active unpacks
2021-06-14 19:13:25.729 +02:00 [DBG] Processing 0 completed downloads
2021-06-14 19:13:25.729 +02:00 [DBG] Processing 0 completed extractions
2021-06-14 19:13:25.729 +02:00 [DBG] Found 1 torrents
2021-06-14 19:13:25.729 +02:00 [DBG] Updating torrent info from RealDebrid
2021-06-14 19:13:26.215 +02:00 [DBG] Finished updating torrent info from RealDebrid, next update in 1 seconds
2021-06-14 19:13:26.215 +02:00 [DBG] Found 0 torrents queued for download
2021-06-14 19:13:26.215 +02:00 [DBG] Found 0 torrents queued for unpacking
2021-06-14 19:13:26.215 +02:00 [DBG] Torrent VGK2QPIWQ4HXS selecting files
2021-06-14 19:13:26.216 +02:00 [DBG] Torrent VGK2QPIWQ4HXS determining which files are already available
2021-06-14 19:13:26.378 +02:00 [ERR] Unexpected error occurred in TorrentDownloadManager.Tick
System.Exception: Unable to deserialze Real Debrid API response to Dictionary`2. Response was: {
	"e4ca350a7a4977110f36e75f08ff642560252eb0": [
		
	]
}
 ---> Newtonsoft.Json.JsonSerializationException: Cannot deserialize the current JSON array (e.g. [1,2,3]) into type 'System.Collections.Generic.Dictionary`2[System.String,System.Collections.Generic.List`1[System.Collections.Generic.Dictionary`2[System.String,RDNET.TorrentInstantAvailabilityFile]]]' because the type requires a JSON object (e.g. {"name":"value"}) to deserialize correctly.
To fix this error either change the JSON to a JSON object (e.g. {"name":"value"}) or change the deserialized type to an array or a type that implements a collection interface (e.g. ICollection, IList) like List<T> that can be deserialized from a JSON array. JsonArrayAttribute can also be added to the type to force it to deserialize from a JSON array.
Path 'e4ca350a7a4977110f36e75f08ff642560252eb0', line 2, position 46.
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.EnsureArrayContract(JsonReader reader, Type objectType, JsonContract contract)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateList(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, Object existingValue, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateDictionary(IDictionary dictionary, JsonReader reader, JsonDictionaryContract contract, JsonProperty containerProperty, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonSerializerSettings settings)
   at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonSerializerSettings settings)
   at RDNET.RdNetClient.Get[T](String url, Boolean repeatRequest)
   --- End of inner exception stack trace ---
   at RDNET.RdNetClient.Get[T](String url, Boolean repeatRequest)
   at RDNET.RdNetClient.GetAvailableFiles(String id)
   at RdtClient.Service.Services.Torrents.GetAvailableFiles(String hash) in /appserver/server/RdtClient.Service/Services/Torrents.cs:line 181
   at RdtClient.Service.Services.TorrentRunner.Tick() in /appserver/server/RdtClient.Service/Services/TorrentRunner.cs:line 380
   at RdtClient.Service.Services.TaskRunner.ExecuteAsync(CancellationToken stoppingToken) in /appserver/server/RdtClient.Service/Services/TaskRunner.cs:line 36
2021-06-14 19:13:28.382 +02:00 [DBG] TorrentRunner Tick Start
```

**Comment by rogerfar** [comment]
It looks like the response from Real Debrid is a bit malformed:

```
{
	"e4ca350a7a4977110f36e75f08ff642560252eb0": [
		
	]
}
```

**Comment by minus1kelvin** [comment]
Yes still happens on occasion for me too

**Comment by rogerfar** [comment]
This should be really fixed in 1.7.8 now.
